src/libs/actions/PersistedRequests.ts

```diff
@@ -54,8 +82,52 @@ function update(oldRequestIndex: number, newRequest: Request) {
     Onyx.set(ONYXKEYS.PERSISTED_REQUESTS, requests);
 }
 
+function updateOngoingRequest(newRequest: Request) {
+    ongoingRequest = newRequest;
+
+    if (newRequest.persistWhenOngoing) {
+        Onyx.set(ONYXKEYS.PERSISTED_ONGOING_REQUESTS, newRequest);
+    }
+}
+
+function processNextRequest(): Request | null {
+    if (ongoingRequest) {
+        Log.info(`Ongoing Request already set returning same one ${ongoingRequest.commandName}`);
+        return ongoingRequest;
+    }
+
+    // You must handle the case where there are no requests to process
+    if (persistedRequests.length === 0) {
+        throw new Error('No requests to process');
+    }
+
+    ongoingRequest = persistedRequests.shift() ?? null;
+
+    if (ongoingRequest && ongoingRequest.persistWhenOngoing) {
+        Onyx.set(ONYXKEYS.PERSISTED_ONGOING_REQUESTS, ongoingRequest);
+    }
+
+    return ongoingRequest;
+}
+
+function rollbackOngoingRequest() {
+    if (!ongoingRequest) {
+        return;
+    }
+
+    // Prepend ongoingRequest to persistedRequests
+    persistedRequests = [ongoingRequest, ...persistedRequests];
```

Comment:
Pretty sure this will be more efficient and avoid copying all the persisted requests.

```suggestion
    persistedRequests.unshift(ongoingRequest);
```

Reply:
Good catch, was replacing to then and forgot this one